### PR TITLE
Update validate files

### PR DESCRIPTION
**Pull request by ronykoz**
## Status
Ready

## Description
Update the way validate_files work, unite the modified and added files logic

### CODE REVIEW

File: demisto_sdk/validation/file_validator.py [hunk 1]
```diff
@@ -244,214 +253,45 @@ def is_valid_release_notes(self, file_path):
...
-            modified_files (set): A set of the modified files in the current branch.
-        """
-        for file_path in modified_files:
-            old_file_path = None
-            if isinstance(file_path, tuple):
-                old_file_path, file_path = file_path
-
+    def validate_files(self, affected_files, is_added=False):
+        REGEX_TO_VALIDATOR = {
+            TEST_PLAYBOOK_REGEX: lambda x: print('Not validating Test playbook'),
```
Comment (yaakovi) [inline demisto_sdk/validation/file_validator.py]: actually, we need to validate version (-1) in test playbooks: https://github.com/demisto/etc/issues/19726

File: demisto_sdk/validation/file_validator.py [hunk 2]
```diff
@@ -244,214 +253,45 @@ def is_valid_release_notes(self, file_path):
...
+            YML_INTEGRATION_REGEXES: self.validate_integration_yml,
+            YML_BETA_INTEGRATIONS_REGEXES: self.validate_beta_integration,
+            IMAGE_REGEX: self.validate_image,
+            REPUTATION_REGEX: self.validate_reputation_file,
+            SCRIPT_REGEX: self.validate_script,
+            PACKAGE_SCRIPTS_REGEXES: self.validate_script,  # todo: check this out
```
Comment (yaakovi) [inline demisto_sdk/validation/file_validator.py]: why scripts are separated and integrations are joined?

File: demisto_sdk/validation/file_validator.py [hunk 3]
```diff
@@ -244,214 +253,45 @@ def is_valid_release_notes(self, file_path):
...
+            PLAYBOOKS_REGEXES_LIST: self.validate_playbook,
+            JSON_INDICATOR_AND_INCIDENT_FIELDS: self.validate_indicator_and_incident_fields,
+            JSON_ALL_LAYOUT_REGEXES: self.validate_layout
```
Comment (yaakovi) [inline demisto_sdk/validation/file_validator.py]: add comment for un-tested files:
Classifiers,
```suggestion
            JSON_ALL_LAYOUT_REGEXES: self.validate_layout,
            # TODO
            # Dashboards,
            # Incident Types,
            # Reports, 
```

File: demisto_sdk/validation/file_validator.py [hunk 4]
```diff
@@ -244,214 +253,45 @@ def is_valid_release_notes(self, file_path):
...
-
-            elif checked_type(file_path, JSON_ALL_LAYOUT_REGEXES):
-                layout_validator = LayoutValidator(structure_validator)
-                if not layout_validator.is_valid_layout():
-                    self._is_valid = False
-
-            elif 'CHANGELOG' in file_path:
-                self.is_valid_release_notes(file_path)
+            self.validate_id_set_data(file_path, is_added)
+            for key in REGEX_TO_VALIDATOR.keys():
```
Comment (yaakovi) [inline demisto_sdk/validation/file_validator.py]:
```suggestion
            for key in REGEX_TO_VALIDATOR:
```

File: demisto_sdk/validation/file_validator.py [hunk 5]
```diff
@@ -244,214 +253,45 @@ def is_valid_release_notes(self, file_path):
...
+                if matched_regex in key:
+                    if 'is_added' in signature(REGEX_TO_VALIDATOR[key]).parameters.keys():
+                        REGEX_TO_VALIDATOR[key](file_path=file_path,
```
Comment (yaakovi) [inline demisto_sdk/validation/file_validator.py]: what validator doesn't have that flag?
shouldn't we add it to all validators?

File: demisto_sdk/validation/file_validator.py [hunk 6]
```diff
@@ -244,214 +253,45 @@ def is_valid_release_notes(self, file_path):
...
+                    else:
+                        REGEX_TO_VALIDATOR[key](file_path=file_path)
 
-            elif checked_type(file_path, [REPUTATION_REGEX]):
-                print_color(
-                    F'Skipping validation for file {file_path} since no validation is currently defined.',
-                    LOG_COLORS.YELLOW)
+                    break
 
             else:
```
Comment (yaakovi) [inline demisto_sdk/validation/file_validator.py]: nice

File: demisto_sdk/validation/file_validator.py [hunk 7]
```diff
@@ -256,15 +255,18 @@ def is_valid_release_notes(self, file_path):
     def validate_files(self, affected_files, is_added=False):
         REGEX_TO_VALIDATOR = {
```
Comment (yaakovi) [inline demisto_sdk/validation/file_validator.py]: `StructureValidator` can be created here and pass to each of the functions.

Comment (anara123) [reply demisto_sdk/validation/file_validator.py]: no need to put too much of effort on it
this written in a bad way
will refactor some of the code anyway

**Comment by ronykoz** [comment]
@anara123 what are we doing with this?

**Comment by ronykoz** [comment]
closing this one for now, we had too many changes there

**Pull request closed without merging**